sdk/identity/azure-identity/samples/azure-aad-auth-with-redis-py.md

```diff
@@ -0,0 +1,107 @@
+## Azure Cache for Redis: Azure AD with redis client library
+
+### Table of contents
+
+- [Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-hello-world)
+- [Authenticate with Azure AD - Handle Re-authentication](#authenticate-with-azure-ad-handle-re-authentication)
+
+#### Samples guidance
+
+Familiarity with the [redis](https://pypi.org/project/redis/) and [azure-identity for Python](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/identity/azure-identity) client libraries is assumed.
+
+[Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-hello-world)
+This sample is recommended for users getting started to use Azure AD authentication with Azure Cache for Redis.
+
+[Authenticate with Azure AD - Handle Re-Authentication](#authenticate-with-azure-ad-handle-re-authentication)
+This sample is recommended to users looking to build long-running applications that would like to handle re-authenticating with Azure AD upon token expiry.
+
+**Note:** The below sample uses the Azure Identity library's `ClientSecretCredential`. The credential can be replaced with any of the other `TokenCredential` implementations offered by the azure-identity library.
+
+#### Authenticate with Azure AD: hello world
+
+This sample is intended to assist in authenticating with Azure AD via the redis client library. It focuses on displaying the logic required to fetch an Azure AD access token and to use it as password when setting up the redis instance.
+
+##### Migration guidance
+
+When migrating your existing application code, you need to replace the password input with the Azure AD token.
+Integrate the logic in your application code to fetch an Azure AD access token via the azure-identity library as shown below and replace it with the password configuring/retrieving logic in your application code.
+
+```py
+import redis
+from azure.identity import DefaultAzureCredential
+
+scope = "https://*.cacheinfra.windows.net:10225/appid/.default"  # The scope will be changed for AAD Public Preview
```

Comment:
I'm a bit out of the loop, so the comment on this line doesn't make much sense to me. Does this mean that the displayed scope is what needs to be used today, but that AAD will be changing the scope as part of a public preview?

Reply:
Right. This was requested by the feature team to explicitly point out this scope is for temporary use. They will change it for public review.